### ISSUE TITLE
Inconsistent Mac only error handling

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#96]
We have two cases where we have mac only error handling. In one case we kill the game, in the other we ignore. They should be consistent, it's an exceedingly weak argument to suggest that two cosmetic game benefits should have different error handling.

This conversation started in PR #87 where @veqryn  suggested we *need* to see these errors to get users to report them, otherwise they would be unfixed. My difference of opinion is that end users *do not* need to see errors and a game crash for cosmetic problems, and that we should have mac release testers look for these kinds of issues (which reveals a gap today, for the errors which we print to standard out, we will not be catching these, and we do that quite a lot).

**Comment by veqryn** [comment]
I am only ok with removing the fatal errors for mac's on cosmetic issues, if we get proper testing and errors in place for macs.
Maybe a test in GameRunner2 instead, that immediately gives a fatal if the mac build was not done successfully?
Until we have something better in place, all you are doing is hiding something that caused major issues.

**Comment by DanVanAtta** [comment]
The icon being the tripleA image vs the Java application image is cosmetic. Is there another problem that was caused when we failed to find the icon image?

A test in GameRunner2 would still not help unless you were on a MacOS, or unless you complicated the test to and re-ran simulating different OS. Part of the trick is to find ways to get rid of the OS specific logic.

Furthermore, the test would probably be run by travis, and would have to be run on the packaged artifact after it was exploded and installed to be sure we packaged everything with correct paths. That is not very practical, and impossible with Travis since we can't yet run that with a Mac docker image.
